modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/database/H2TreeIndexBase.java

```diff
@@ -101,13 +111,26 @@ static int computeInlineSize(List<InlineIndexColumn> inlineIdxs, int cfgInlineSi
         int size = 0;
 
         for (InlineIndexColumn idxHelper : inlineIdxs) {
-            if (idxHelper.size() <= 0) {
-                size = propSize;
-                break;
+            // for variable types - default variable size, for other types - type's size + byte type
+            int sizeInc = idxHelper.size() <= 0 ? IGNITE_VARIABLE_TYPE_DEFAULT_INDEX_SIZE : idxHelper.size() + 1;
```

Comment:
why we use LE or EQ to zero? I understand that is copy past from previous code, however why we should reserve DEFAULT_SIZE for zero size?
@korlov42 could we have zero type at all? may be we need additional assertion in AbstractInlineIndexColumn constructor?

Reply:
For me support of 0-sized types seems incorrect. But I also don't see where some assertion can be added. Don't think that computeInlineSize(), which is being called for customer indexes, should check our inner implementation of sql type. Size is a hardcoded value and it is being set by ourselves for the each type. What if I will just add some description for the property that will contain expected values?

Reply:
Assertion should be added to constructor of AbstractInlineIndexColumn